test/cpp/end2end/end2end_test.cc

```diff
@@ -2175,6 +2182,7 @@ std::vector<TestScenario> CreateTestScenarios(bool use_proxy,
 #endif
 
   if (test_secure) {
+    EnableSpiffeCredentials(GetCredentialsProvider());
```

Comment:
It seems all other credential types are obtained via `GetCredentialsProvider()->GetSecureCredentialsTypeList()`. Can we do the same for TLS?

Reply:
Done. This caused other e2e tests (e.g. `PortSharingEnd2endTest`) to start using the SPIFFE credentials, which necessitated some changes to `tls_test_credentials.{h,cc}`. Namely, I needed to introduce the `TlsThread` and `TlsData` objects. PTAL when I post the next commit.